File: A2OJ/CodeforcesDiv2_B/141.cpp
# include <bits/stdc++.h>

using namespace std;

# define looptc					int t, c=0; cin >> t; while(t--&&++c)
# define ln                     "\n"
# define ll						long long

int main() {
    ios_base::sync_with_stdio(0); cin.tie(0); cout.tie(0);
    freopen("_input.txt", "r+", stdin);

    int na, ma, nb, mb, x;
    vector<vector<int>> a, b;
    vector<int> t;
    cin >> na >> ma;
    for(int i=0; i<na; ++i) {
        for(int j=0; j<ma; ++j)
            cin >> x, t.push_back(x);
        a.push_back(t);
        t.clear();
    }

    for(int i=0; i<nb; ++i) {
        for(int j=0; j<mb; ++j)
            cin >> x, t.push_back(x);
        b.push_back(t);
        t.clear();
    }

    cout << endl;
    return 0;
}

File: A2OJ/CodeforcesDiv2_B/141.py
import sys
#sys.stdin = open('_input.txt')

na, ma = list(map(int, input().split()))
a = []
for _ in range(na):
    x = input()
    a.append([1 if x[i] == '1' else 0 for i in range(len(x))])

nb, mb = list(map(int, input().split()))
b = []
for _ in range(nb):
    x = input()
    b.append([1 if x[i] == '1' else 0 for i in range(len(x))])

m = 0
res = [0, 0]
for x in range(1-na, nb):
    for y in range(1-ma, mb):
        s = 0
        for i in range(na):
            for j in range(ma):
                if 0 <= i+x < nb and 0 <= j+y < mb:
                    s += a[i][j] * b[i+x][j+y]
        if s > m:
            res = [x, y]
            m = s

print(' '.join(map(str, res)))
